### ISSUE TITLE
Please add a priority for each Link so that LinkExtractor can do crawl focusing/priority

### ISSUE THREAD

**Issue by danizen** [Norconex/collector-http#344]
Some domains may be more important than others, and the link text may indicate that one link may be more important than others.   I think the queue would also have to be changed so that it uses either multiple lists for different priorities or is implemented as a priority queue.

At least I'm not asking for this to be a distributed crawler; I don't think I need that.

**Comment by essiembre** [comment]
You mean when you are crawling multiple domains, you want to "hit" some domains more often than others (crawl faster)? This can be accomplish already.  Here are two methods.  

First, you can create multiple crawler/collector configs, one for each group of domains you want to treat differently, with different ``delay`` settings.  This may not be the most convenient if you have tons domains you want treat differently.

Maybe more flexible, you can use [ReferenceDelayResolver](https://www.norconex.com/collectors/collector-http/latest/apidocs/com/norconex/collector/http/delay/impl/ReferenceDelayResolver.html).  With several threads, you can make sure they are not all super busy at the same time.  Set the default delay to what you want most of your domain crawls to be... then use URL patterns to slow/speed some.  You likely want to set the scope to ``site``.

**Comment by danizen** [comment]
That looks reasonable.    

Another question, rather than create lots of issues.   I have problems that stem from not understanding Elastic Search terribly well (not as well as Solr).   I got everything to work yesterday by building a 3-node ES cluster on my desktop, and today I attempted to simplify by having no replicas and 1 shard and use a simple 1-node ES cluster.   I'm still getting time-outs and Exceptions when norconex-committer-elasticsearch reaches the commit point.

Then, I redid config/elasticsearch.yml for 3 and tried again, and I am again getting issues.   What is a good link to basically understanding where elasticsearch keeps its data and meta-data.  Just removing the data directory was not enough.

Also, is Norconex doing anything special that make this less likely to work, I mean other than using the TCP transport that will be faster and handle single node failures if all is well architected.
